pkg/vpcmodel/subnetsConnectivity.go

```diff
@@ -0,0 +1,289 @@
+			}
+		}
+	}
+	if subnetsConnectivityFromACLresources == nil {
+		return nil, errors.New("missing connectivity results from NACL resources")
+	}
+
+	pgwConns := map[string]ConfigBasedConnectivityResults{}
+	for _, r := range c.RoutingResources {
+		if r.Kind() == "PublicGateway" {
+			conns := r.ConnectivityMap()
+			for subnetCidr, pgwConnsPerCidr := range conns {
+				pgwConns[subnetCidr] = pgwConnsPerCidr
+			}
+		}
+	}
+
+	// convert to subnet-based connectivity result
+	subnetsConnectivity := map[string]*ConfigBasedConnectivityResults{}
+	for subnetCidrStr, ipBasedConnectivity := range subnetsConnectivityFromACLresources {
+		subnetNodeSet, err := c.subnetCidrToSubnetElem(subnetCidrStr)
+		if err != nil {
+			return nil, err
+		}
+
+		// create and update configBasedConns according to relevant router (pgw) resources
+		subnetHasPGW := false
+		if _, ok := pgwConns[subnetCidrStr]; ok {
+			subnetHasPGW = true
+		}
+		if !includePGW {
+			subnetHasPGW = true // do not limit connectivity to external nodes only if has actual PGW
+		}
+		configBasedConns, err := c.ConvertIPbasedToSubnetBasedResult(ipBasedConnectivity, subnetHasPGW)
+		if err != nil {
+			return nil, err
+		}
+
+		subnetsConnectivity[subnetNodeSet.Name()] = configBasedConns
+	}
+
+	res := &VPCsubnetConnectivity{AllowedConns: subnetsConnectivity, cloudConfig: c}
+
+	res.printAllowedConns()
+	// get combined connections from subnetsConnectivity
+	if err := res.computeAllowedConnsCombined(); err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+func (v *VPCsubnetConnectivity) computeAllowedConnsCombined() error {
+	v.AllowedConnsCombined = map[string]map[string]*common.ConnectionSet{}
+	for subnetNodeSet, connsRes := range v.AllowedConns {
+		for peerNode, conns := range connsRes.IngressAllowedConns {
+			src := peerNode
+			dst := subnetNodeSet
+			combinedConns := conns.Copy()
+
+			// peerNode kind is expected to be Subnet or External
+			peerNodeObj := v.cloudConfig.NameToResource[peerNode]
+			switch concPeerNode := peerNodeObj.(type) {
+			case NodeSet:
+				egressConns := v.AllowedConns[concPeerNode.Name()].EgressAllowedConns[subnetNodeSet]
+				combinedConns = combinedConns.Intersection(egressConns)
+			case *ExternalNetwork:
+				// do nothing
+			default:
+				return errors.New(errUnexpectedTypePeerNode)
+			}
+			if _, ok := v.AllowedConnsCombined[src]; !ok {
+				v.AllowedConnsCombined[src] = map[string]*common.ConnectionSet{}
+			}
+			v.AllowedConnsCombined[src][dst] = combinedConns
+		}
+		for peerNode, conns := range connsRes.EgressAllowedConns {
+			src := subnetNodeSet
+			dst := peerNode
+			combinedConns := conns
+
+			// peerNode kind is expected to be Subnet or External
+			peerNodeObj := v.cloudConfig.NameToResource[peerNode]
+			switch concPeerNode := peerNodeObj.(type) {
+			case NodeSet:
+				ingressConns := v.AllowedConns[concPeerNode.Name()].IngressAllowedConns[subnetNodeSet]
+				combinedConns = combinedConns.Intersection(ingressConns)
```

Comment:
Aren't we doing double work here? We should already have this <src, dst> pair in the map with the right ConnectionSet.
Can't we just `continue` if peerNode is not external?

Reply:
right, good catch! fixed here, and in `vpcConnectivity.go` as well